### DISCUSSION TITLE
⚠️ v14 Release allow login without auth

### DISCUSSION THREAD

**Discussion by Romancha** [wg-easy/wg-easy#1478] [General]
Hi wg-easy team,

I've encountered the same issue described in [this issue](https://github.com/wg-easy/wg-easy/issues/1473). I noticed that you've implemented the use of password hashing, which is a great improvement—thank you for that!

However, in release 14, there was a bug where it was possible to access the page without entering a password. These things happen, and that’s understandable.

For reference, the initial build of `ghcr.io/wg-easy/wg-easy:14` was created on 2024-08-09, bug is reproduced:

```
        "Id": "sha256:1067d2f56e2c4d7b8fc18b053b48591158f4cd9deaf011a825cbc37c96488bbf",
        "RepoTags": [
            "ghcr.io/wg-easy/wg-easy:14"
        ],
        "RepoDigests": [
            "ghcr.io/wg-easy/wg-easy@sha256:f6c2587da3fdc9d5927743fb4a92cff42d690b182396ffd0cdad034becb41053"
        ],
        "Parent": "",
        "Comment": "buildkit.dockerfile.v0",
        "Created": "2024-08-09T19:33:11.489509931Z",
```

Now `ghcr.io/wg-easy/wg-easy:14` builded on 2024-09-04, bug is not reproduced:

```
        "Id": "sha256:d8ad20914b334d7ebcd7dfa6df055e2228ee4d98fc2b208d52266028bf680f89",
        "RepoTags": [
            "ghcr.io/wg-easy/wg-easy:14"
        ],
        "RepoDigests": [
            "ghcr.io/wg-easy/wg-easy@sha256:66352ccb4b5095992550aa567df5118a5152b6ed31be34b0a8e118a3c3a35bf5"
        ],
        "Parent": "",
        "Comment": "buildkit.dockerfile.v0",
        "Created": "2024-09-04T16:44:04.981049662Z",
```

My main concern is about the silent update to the 14 release. Why was the decision made to replace the version under the same tag without notifying the community?

Wouldn't it be better to release a hotfix, such as version 14.1, with a clear message to users that this is an important update? I believe it’s crucial to ensure users are aware, as many may still have their panels exposed without realizing it.

I apologize if my message comes across as too direct, but I believe this is a crucial matter.

Thanks again for all your work! I hope you will consider this suggestion.

**Comment by kaaax0815** [comment]
Related https://github.com/wg-easy/wg-easy/pull/1350 & https://github.com/wg-easy/wg-easy/issues/1269

Currently, we don’t follow Semantic Versioning (Semver). This limits our ability to increment version numbers like you mentioned without actually incrementing to the next "major" (it would have been v15 either way as it is a breaking change). Therefore we opted to include the hotfix with the original tag.

We will switch to Semver with #1333. However, even with this change, updating the software will still replace the existing tag (e.g., 14) with a new version[^1]. The key difference is that we will now also publish a Release that details the changes made.

[^1]: Example: if 14.1.0 is released. tag 14, 14.1 and 14.1.0 points to 14.1.0. With release 14.1.1. tag 14 and 14.1 would be updated to point to 14.1.1
